File: codility/Flags.js
function solution(A) {
    // Implement your solution here[1,3,5,10] [2, 2, 5]
    // 1 5 9
    const pArr = [];
    for (let i = 1; i < A.length - 1; i++) {
        if (A[i - 1] < A[i] && A[i] > A[i + 1]) {
            pArr.push(i);
        }
    }
    if (pArr.length === 0) {
        return 0;
    }
    // console.log(pArr)
    let k = pArr.length;
    while (!calc(pArr, k)) {
        k--;
    }
    function calc(pArr, k) {
        let lastIndex = pArr[0];
        const len = pArr.length;
        if ((pArr[len - 1] - pArr[0]) / (k - 1) < k) {
            return false;
        }
        let counter = 1;
        for (let i = 1; i < len; i++) {
            if (pArr[i] - lastIndex >= k) {
                counter++;
                lastIndex = pArr[i];
                if (counter === k) {
                    break;
                }
            }
        }
        return counter === k;
    }
    return k;
}

File: codility/python/Flags.py
def solution(A):
    # Implement your solution here
    peaks = []
    for i in range(1,len(A)-1):
        if A[i] > A[i-1] and A[i] > A[i+1]:
            peaks.append(i)

    # print(peaks)
    if len(peaks) == 0:
        return 0
    k = len(peaks)
    for i in range(k,0,-1):
        if i * (i - 1) > peaks[-1] - peaks[0]:
            continue
        last_peak = peaks[0]
        count = 1
        for j in range(1, k):
            if peaks[j] - last_peak >= i:
                count += 1
                last_peak = peaks[j]
                if count == i:
                    return i
    return 1